File: ios/Theme/Common/CommonStyle.swift
//
//  CommonSwift.swift
//  Pods
//
//  Created by Daniel Sansão Araldi on 15/08/25.
//

public class CommonStyle {
  private let theme: NSDictionary?
  private let color: Color
  private let linearGradient: LinearGradient
  
  init(key: String) {
    self.theme = Style().getTarget(key)
    self.color = Color()
    self.linearGradient = LinearGradient()
  }
  
  init(theme: NSDictionary?, key: String) {
    self.theme = Style().getTarget(theme, key: key)
    self.color = Color()
    self.linearGradient = LinearGradient()
  }
  
  func getBackgroundColor() -> UIColor {
    return self.color.getColor(self.theme, key: "backgroundColor")
  }

  func getBackgroundColor(defaultColor: String) -> UIColor {
    return self.color.getColor(self.theme, key: "backgroundColor", defaultColor: defaultColor)
  }
  
  func getBackgroundColors() -> [UIColor] {
    return self.linearGradient.getLinearGradient(self.theme, key: "backgroundColors")
  }
  
  func getGradientBackgroundColors() -> CAGradientLayer {
    return self.linearGradient.getGradientLayer(self.theme, key: "backgroundColors")
  }

  func getForegroundColor() -> UIColor {
      return self.color.getColor(self.theme, key: "foregroundColor")
  }

  func getForegroundColor(defaultColor: String) -> UIColor {
      return self.color.getColor(self.theme, key: "foregroundColor", defaultColor: defaultColor)
  }

  func getTextBackgroundColor() -> UIColor {
    return self.color.getColor(self.theme, key: "textBackgroundColor", defaultColor: "#026ff4")
  }
}

File: examples/flutterapp/android/app/src/main/java/com/azifacemobile/theme/abstracts/CommonStyle.java
package com.azifacemobile.theme.abstracts;

import com.azifacemobile.theme.Color;
import com.azifacemobile.utils.Theme;

import org.json.JSONObject;

public abstract class CommonStyle {
  private final JSONObject theme;
  private final Color color;

  public CommonStyle(String key) {
    this.theme = new Theme().getTarget(key);
    this.color = new Color();
  }

  public CommonStyle(JSONObject theme, String key) {
    this.theme = new Theme().getTarget(theme, key);
    this.color = new Color();
  }

  public int getBackgroundColor() {
    return this.color.getColor(this.theme, "backgroundColor");
  }

  public int getBackgroundColor(String defaultColor) {
    return this.color.getColor(this.theme, "backgroundColor", defaultColor);
  }

  public int getForegroundColor() {
    return this.color.getColor(this.theme, "foregroundColor");
  }

  public int getForegroundColor(String defaultColor) {
    return this.color.getColor(this.theme, "foregroundColor", defaultColor);
  }

  public int getTextBackgroundColor() {
    return this.color.getColor(this.theme, "textBackgroundColor", "#026ff4");
  }
}
